modules/openxr/extensions/openxr_vulkan_extension.cpp

```diff
@@ -74,25 +80,22 @@ void OpenXRVulkanExtension::on_instance_created(const XrInstance p_instance) {
 	ERR_FAIL_NULL(openxr_api);
 
 	// Obtain pointers to functions we're accessing here, they are (not yet) part of core.
-	result = xrGetInstanceProcAddr(p_instance, "xrGetVulkanGraphicsRequirements2KHR", (PFN_xrVoidFunction *)&xrGetVulkanGraphicsRequirements2KHR_ptr);
-	if (XR_FAILED(result)) {
-		print_line("OpenXR: Failed to xrGetVulkanGraphicsRequirements2KHR entry point [", openxr_api->get_error_string(result), "]");
-	}
-
-	result = xrGetInstanceProcAddr(p_instance, "xrCreateVulkanInstanceKHR", (PFN_xrVoidFunction *)&xrCreateVulkanInstanceKHR_ptr);
-	if (XR_FAILED(result)) {
-		print_line("OpenXR: Failed to xrCreateVulkanInstanceKHR entry point [", openxr_api->get_error_string(result), "]");
-	}
+	result = openxr_api->get_instance_proc_addr("xrGetVulkanGraphicsRequirements2KHR", (PFN_xrVoidFunction *)&xrGetVulkanGraphicsRequirements2KHR_ptr);
```

Comment:
Should we use the macro's here as well?

Reply:
@BastiaanOlij @kisg  _openxr_vulkan_extension.h_ contains the following comment:

```
// Forward declare these so we don't need OpenXR headers where-ever this is included
// Including OpenXR at this point gives loads and loads of compile issues especially
// on Windows because windows.h is EVIL and really shouldn't be included outside of platform
// but we really don't have a choice in the matter

struct XrGraphicsRequirementsVulkanKHR;
struct XrVulkanInstanceCreateInfoKHR;
struct XrVulkanGraphicsDeviceGetInfoKHR;
struct XrVulkanDeviceCreateInfoKHR;
struct XrGraphicsBindingVulkanKHR;
```

Does this still hold true? Also, should the same apply to _openxr_fb_passthrough_extension_wrapper.h_? In Godot 3.x, the header of `XRFbPassthroughExtensionWrapper` already depends on OpenXR. If this restriction still applies, we might need to complicate the macros a bit so that we can separate the declaration and definition of the wrapper functions.

Reply:
To be honest, it was needed in the earlier implementations related to how we did the Vulkan overrules and this header was being included within the Vulkan driver. That caused all the mean stuff.

Seeing we now nicely isolate things and introduced the `VulkanHooks` class making the Vulkan driver completely unaware of the internals involved here, I think it's now safe to move `#include "../openxr_api.h"` from the cpp file to the header file and getting all the needed includes.